client/app/lib/api.ts

```diff
@@ -45,18 +62,303 @@ export async function fetchMovies(limit: number = 50): Promise<Movie[]> {
+    };
+  } catch (error) {
+    console.error('Error creating movie:', error);
+    return { 
+      success: false, 
+      error: 'Network error occurred while creating movie' 
+    };
+  }
+}
+
+/**
+ * Create multiple movies in a batch operation
+ */
+export async function createMoviesBatch(moviesData: Omit<Movie, '_id'>[]): Promise<{ success: boolean; error?: string; insertedCount?: number; insertedIds?: string[] }> {
+  try {
+    const response = await fetch(`${API_BASE_URL}/api/movies/batch`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(moviesData),
+    });
+
+    const result = await response.json();
+
+    if (!response.ok) {
+      return { 
+        success: false, 
+        error: result.error || `Failed to create movies: ${response.status}` 
+      };
+    }
+
+    if (!result.success) {
+      return { 
+        success: false, 
+        error: result.error || 'API returned error response' 
+      };
+    }
+
+    return { 
+      success: true, 
+      insertedCount: result.data.insertedCount,
+      insertedIds: result.data.insertedIds ? Object.values(result.data.insertedIds) : []
+    };
+  } catch (error) {
+    console.error('Error creating movies batch:', error);
+    return { 
+      success: false, 
+      error: 'Network error occurred while creating movies' 
+    };
+  }
+}
+
+/**
+ * Delete multiple movies in a batch operation
+ */
+export async function deleteMoviesBatch(movieIds: string[]): Promise<{ success: boolean; error?: string; deletedCount?: number }> {
+  try {
+    // Create filter to match the movie IDs
+    // Note: The server will handle ObjectId conversion
+    const filter = {
+      _id: {
+        $in: movieIds
+      }
+    };
+
+    const response = await fetch(`${API_BASE_URL}/api/movies`, {
+      method: 'DELETE',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ filter }),
+    });
+
+    const result = await response.json();
+
+    if (!response.ok) {
+      return { 
+        success: false, 
+        error: result.error || `Failed to delete movies: ${response.status}` 
+      };
+    }
+
+    if (!result.success) {
+      return { 
+        success: false, 
+        error: result.error || 'API returned error response' 
+      };
+    }
+
+    return { 
+      success: true, 
+      deletedCount: result.data.deletedCount
+    };
+  } catch (error) {
+    console.error('Error deleting movies batch:', error);
+    return { 
+      success: false, 
+      error: 'Network error occurred while deleting movies' 
+    };
+  }
+}
+
+/**
+ * Update multiple movies in a batch operation
+ */
+export async function updateMoviesBatch(movieIds: string[], updateData: Partial<Movie>): Promise<{ success: boolean; error?: string; matchedCount?: number; modifiedCount?: number }> {
+  try {
+    // Create filter to match the movie IDs
```

Comment:
```suggestion
    // Create filter to match the movie IDs
    // Note: The server will handle ObjectId conversion
```
applicable here as well?